sklearn/metrics/regression.py

```diff
@@ -134,14 +147,28 @@ def mean_absolute_error(y_true, y_pred, sample_weight=None):
     >>> y_pred = [[0, 2], [-1, 2], [8, -5]]
     >>> mean_absolute_error(y_true, y_pred)
     0.75
-
+    >>> mean_absolute_error(y_true, y_pred, output_weights=None)
+    array([ 0.5,  1. ])
+    >>> mean_absolute_error(y_true, y_pred, output_weights=[0.3, 0.7])
+    ... # doctest: +ELLIPSIS
+    0.849...
     """
-    y_type, y_true, y_pred = _check_reg_targets(y_true, y_pred)
-    return np.average(np.abs(y_pred - y_true).mean(axis=1),
-                      weights=sample_weight)
-
-
-def mean_squared_error(y_true, y_pred, sample_weight=None):
+    y_type, y_true, y_pred, output_weights = _check_reg_targets(
+        y_true, y_pred, output_weights)
+    output_errors = np.average(np.abs(y_pred - y_true),
+                               weights=sample_weight, axis=0)
+    if output_weights is None:
+        return output_errors
+    elif output_weights == 'uniform':
+        # pass None as weights to np.average: uniform mean
+        output_weights = None
+
+    return np.average(output_errors, weights=output_weights)
+
+
+def mean_squared_error(y_true, y_pred,
+                       output_weights='uniform',
+                       sample_weight=None):
```

Comment:
output_weights should be put after sample_weight (in the function signature and in the docstring) and renamed to a singular: output_weight. In addition, we should name it in some way that makes it obvious that this is only for multi-output. Maybe "multioutput_weight". Open to other suggestions.